netket/experimental/driver/tdvp_common.py

```diff
@@ -103,37 +106,54 @@ def __init__(
         self._dw = None  # type: PyTree
         self._last_qgt = None
         self._integrator = None
-        self._integrator_constructor = None
 
         self._odefun = HashablePartial(odefun_host_callback, self.state, self)
 
         self.error_norm = error_norm
-        self.integrator = integrator
+
+        if integrator is not None:
+            warn_deprecation(
+                "Specifying `integrator` when constructing TDVP is deprecated. Please use `solver` instead."
+            )
+            if solver is not None:
+                raise ValueError("Cannot specify both `solver` and `integrator`.")
+            solver = integrator
+        if solver is None:
+            raise ValueError("You need to define the `solver` for the TDVP driver.")
+        self.solver = solver
```

Comment:
This should work properly then...? (Sorry, not used to throwing deprecation warnings ahah).